pkg/controller/registry/resolver/hooks.go

```diff
@@ -0,0 +1,9 @@
+package resolver
+
+// StepResolverInitHook provides a way for the downstream
+// to modify the step resolver at creation time.
+// This is a bit of a hack to enable system constraints downstream
+// without affecting the upstream. We may want to clean this up when
+// either we have a more pluggable architecture; or system constraints
+// come to the upstream
+type StepResolverInitHook func(*OperatorStepResolver) error
```

Comment:
If the injection only occurs in-package, then this interface doesn't need to be public, right?

Reply:
+1 - great catch - ty!